cmake/emu/simics.cmake

```diff
@@ -1,30 +1,42 @@
-# Copyright (c) 2023 Intel Corporation
+# Copyright (c) 2023-2024 Intel Corporation
 #
 # SPDX-License-Identifier: Apache-2.0
 
 find_program(
   SIMICS
   NAMES simics
+  NO_DEFAULT_PATH
+  PATHS ENV SIMICS_PROJECT
+  # Search exactly for the project's autogenerated 'trampoline' script.
   )
 
-zephyr_get(SIMICS_SCRIPT_PATH SYSBUILD GLOBAL)
-if(SIMICS_SCRIPT_PATH)
-  set(SIMICS_SCRIPT ${SIMICS_SCRIPT_PATH})
+if(SIMICS STREQUAL SIMICS-NOTFOUND)
+  message(WARNING "Simics simulator environment is not found at SIMICS_PROJECT:'${SIMICS_PROJECT}'")
 else()
-  set(SIMICS_SCRIPT ${BOARD_DIR}/support/${BOARD}.simics)
-endif()
+  message(STATUS "Found Simics: ${SIMICS}")
 
-get_property(SIMICS_ARGS GLOBAL PROPERTY "BOARD_EMU_ARGS_simics")
+  zephyr_get(SIMICS_SCRIPT_PATH SYSBUILD GLOBAL)
+  if(SIMICS_SCRIPT_PATH)
+    set(SIMICS_SCRIPT ${SIMICS_SCRIPT_PATH})
+  else()
+    set(SIMICS_SCRIPT ${BOARD_DIR}/support/${BOARD}.simics)
+  endif()
 
-add_custom_target(run_simics
-  COMMAND
-  ${SIMICS}
-  -no-gui
-  -no-win
-  ${SIMICS_SCRIPT}
-  ${SIMICS_ARGS}
-  -e run
-  WORKING_DIRECTORY ${APPLICATION_BINARY_DIR}
-  DEPENDS ${logical_target_for_zephyr_elf}
-  USES_TERMINAL
-  )
+  get_property(SIMICS_ARGS GLOBAL PROPERTY "BOARD_EMU_ARGS_simics")
+
+  add_custom_target(run_simics
+    COMMAND
+    ${SIMICS}
+    -no-gui
+    --no-win
+    --batch-mode
+    ${SIMICS_SCRIPT}
+    ${SIMICS_ARGS}
+    $ENV{SIMICS_EXTRA_ARGS}
```

Comment:
Should we use `EXTRA_SIMICS_ARGS` to be aligned with other Zephyr CMake settings, like `EXTRA_CONF_FILE`, `EXTRA_CFLAGS`, etc. ?

Also, using `zephyr_get(EXTRA_SIMICS_ARGS ...)` will allow users to choose freely between sysbuild, CMake var `-DEXTRA_SIMICS_ARGS=<val>`, or environment variable and not force them into using env vars.